sklego/preprocessing.py

```diff
@@ -0,0 +1,48 @@
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+
+class PandasTypeSelector(BaseEstimator, TransformerMixin):
+    """
+    Select columns in a pandas dataframe based on their dtype
+
+    :param include: types to be included in the dataframe
+    :param exclude: types to be exluded in the dataframe
+    """
+    def __init__(self, include, exclude=[]):
+        self.include = include
+        self.exclude = exclude
+
+    def fit(self, X, y=None):
+        """
+        Saves the column names for check during transform
+
+        :param X: pandas dataframe to select dtypes for
+        :param y: not used in this class
+        """
+        self._check_X_for_type(X)
```

Comment:
maybe an explicit `ValueError` when you select columns that do not appear in the dataframe that is brought in here? 

you probably also want to do this during the `.transform()` phase

Reply:
Yes I changed it